### PR TITLE
LF-3421 Days of Week Select component

### PR DESCRIPTION
**Pull request by kathyavini**
**Description**

This PR creates a new component for use with the Repeat Crop Plan view (and other views in the future), a selector for picking days of the week.

Notes:
- The default behaviour of the component is to allow only one selection at a time, but this is configurable with the property `maxSelect`
- It is a controlled component of multiple checkboxes, more or less following the checkboxes example from the [React Hook Form Documentation](https://react-hook-form.com/docs/usecontroller)
-  It can be situated within the React Hook Form using the `<Controller />` component following the pattern used for `<ReactSelect />`

```jsx
<Controller
  control={control}
  name={'days_of_week'}
  rules={{ required: true }}
  render={({ field: { onChange, value } }) => (
    <DaysOfWeekSelect
      onChange={onChange}
      defaultValue={value}
      maxSelect={1}
      errors={getInputErrors(errors, 'days_of_week')}
    />
  )}
/>
```

Jira link: https://lite-farm.atlassian.net/browse/LF-3421

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [x] Other (please explain)

The component is found within Storybook under `Components/DaysOfWeekSelect`

To run the automated tests (play tests):

```bash
pnpm storybook # wait for Storybook to finish loading
pnpm test-storybook src/stories/Form/DaysOfWeekSelect
```

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [x] I have added or updated language tags for text that's part of the UI
- [x] I have added "MISSING" for all new language tags to languages I don't speak
- [x] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Review by SayakaOno — COMMENTED** [review]
It looks great! One thing I was not sure is that all the inputs have the same "name" and how it's going to work with the react-hook-form, but I think the component can be adjusted later :)
Why are there no previous years in the license??

**Comment by iperdomo** [comment]
> Why are there no previous years in the license??

The years is a convention on when the file was created and modified. If the file is new, then the value should only be 2023.
This is how the GNU project suggest to use the copyright header: https://www.gnu.org/licenses/gpl-howto.html#copyright-notice

There is a recent trend to just avoid put any year, e.g.
https://daniel.haxx.se/blog/2023/01/08/copyright-without-years/

**Comment by SayakaOno** [comment]
@iperdomo 
I understand.

The license we have been using is:
```...Copyright 2019, 2020, 2021, 2022, 2023 LiteFarm.org...```
Should we change it to:
```...Copyright 2023 LiteFarm.org...```

Sorry if it was discussed while I was away.

File: packages/webapp/src/components/Form/DaysOfWeekSelect/index.jsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    { label: t('DAYS_OF_WEEK_SELECT.WEDNESDAY'), value: 'Wednesday' },
+    { label: t('DAYS_OF_WEEK_SELECT.THURSDAY'), value: 'Thursday' },
+    { label: t('DAYS_OF_WEEK_SELECT.FRIDAY'), value: 'Friday' },
+    { label: t('DAYS_OF_WEEK_SELECT.SATURDAY'), value: 'Saturday' },
+  ];
+
+  const [selected, setSelected] = useState(defaultValue ? defaultValue : []);
+
+  const handleChange = (e, day) => {
+    let updated = [...selected];
```
[the remark this answers is not in the document]
Comment (kathyavini) [reply packages/webapp/src/components/Form/DaysOfWeekSelect/index.jsx]: Ooops, thanks Sayaka! I think I expected to have to do some mutation but it turned out simpler than that 🙂  I have removed the pointless array duplication.

**Comment by kathyavini** [comment]
Thanks @SayakaOno! And sorry for my unnecessarily complicated attempt to talk about `name` this morning 😅 

To summarize that for the records, the name is not used by the React Hook Form (just the `onChange` works to register the component when used with `<Controller />` as in the PR description above), but I included the name only to keep the default HTML structure for multiple checkboxes.

**Review by SayakaOno — APPROVED** [review]

**Review by iperdomo — APPROVED** [review]

**Pull request merged**